Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -63,14 +56,19 @@ import Foundation
         let btHttp = BTHTTP(authorization: self.authorization)
         http = btHttp
         configurationLoader = ConfigurationLoader(http: btHttp)
-        
+
         super.init()
-        BTAPIClient._analyticsService = BTAnalyticsService(apiClient: self)
         http?.networkTimingDelegate = self
 
         // Kickoff the background request to fetch the config
-        fetchOrReturnRemoteConfiguration { _, _ in
-            // No-op
+        fetchOrReturnRemoteConfiguration { configuration, _ in
+            if let configuration {
+                self.analyticsService = BTAnalyticsService(
```

Comment:
Even with this, if we are moving toward injection of configuration into analytics, we have no guarantee that configuration is fetched before first call to sendAnalytics in tokenize function, for example. (batch sending mitigates it but I think there might still be intermittent issues)

Reply:
We could do what Android does, persist the analytic events in local storage, don't know the detail implementation but I remember seeing that.

Reply:
It's a great callout - I'm probably going to close this PR in favor of something else